### PR TITLE
Feature: Email validation

### PR DESCRIPTION
**Pull request by martinlasek**
The email provided to reset password is not checked on valid format

### CODE REVIEW

**Comment by steffendsommer** [comment]
I will leave this for @siemensikkema as he has better overview of what can be re-used for this new piece of functionality.

File: Sources/JWTKeychain/Controllers/API/APIUserControllerDelegate.swift [hunk 1]
```diff
@@ -57,6 +59,15 @@ open class APIUserControllerDelegate<U: JWTKeychainUser>:
         passwordResetMailer: PasswordResetMailerType
     ) throws -> ResponseRepresentable {
         do {
+
```
Comment (siemensikkema) [inline Sources/JWTKeychain/Controllers/API/APIUserControllerDelegate.swift]: unnecessary newline (?)

File: Sources/JWTKeychain/Controllers/API/APIUserControllerDelegate.swift [hunk 2]
```diff
@@ -57,6 +59,15 @@ open class APIUserControllerDelegate<U: JWTKeychainUser>:
...
+            if let json = request.json {
+                let email: String = try json.get("email")
```
Comment (siemensikkema) [inline Sources/JWTKeychain/Controllers/API/APIUserControllerDelegate.swift]: I forget but isn't there a constant for the key "email" somewhere?

File: Sources/JWTKeychain/Controllers/API/APIUserControllerDelegate.swift [hunk 3]
```diff
@@ -87,4 +100,8 @@ extension APIUserControllerDelegate {
     func status(_ status: String) -> ResponseRepresentable {
         return JSON(["status": .string(status)])
     }
+
+    enum EmailError: Error {
```
Comment (siemensikkema) [inline Sources/JWTKeychain/Controllers/API/APIUserControllerDelegate.swift]: can this be private?

**Review by siemensikkema — APPROVED** [review]

**Pull request merged**